taurus_pyqtgraph/datainspectormodel.py

```diff
@@ -0,0 +1,270 @@
+        :param point_index: index of picked point
+        :param i: the index of current curve
+        """
+
+        # Update items to make sure the older point is not still trigger
+        curve.updateItems()
+        scatterPoint = self._scatter[i].points()[point_index[0]]
+        scatterPoint.setSize(self.trigger_point_size)
+
+    def detach(self):
+        """
+        Removing the :class:`pyqtgraph.InfiniteLine` and
+        :class:`pyqtgraph.TextItem` from the plot
+        """
+        self._resetCustomStyle()
+        self.plot.scene().removeItem(self._v_line)
+        for cloud in self._information_clouds.values():
+            self.plot.scene().removeItem(cloud)
+
+    def _getPointIndex(self, np_curve_data, mouse_point):
+        """
+        Helper method used to checking if the current mouse point coordinates
+        are near the any point's in the curve
+
+        :param np_curve_data: the array with the data (x or y)
+        :param mouse_point: the current coordinates of the mouse (x or y)
+        """
+        point_picker = PointPicker(np_curve_data)
+        roi = self._calculateRoi(self.default_point_size)
+        return point_picker.checkPoint(mouse_point, roi)
+
+    def _calculateRoi(self, point_size, width=1080):
+        """
+        Simply algorithm to calculating roi
+
+        :param point_size: current size of the point
+        :param width: default screen resolution
+        :return: roi : region of interest where checkPoint method
+                       will be searching the point
+        """
+
+        x_axis = self._getXAxis()
+        _min = x_axis.range[0]
+        _max = x_axis.range[1]
+
+        pixel_size = (_max - _min) / width
+        roi = pixel_size * point_size
+
+        return roi
+
+    def _timestampToTime(self, timestamp):
+        """
+        Method used to caste the timestamp from the curve to date
+        in proper format (%Y-%m-%d %H:%M:%S)
+
+        :param timestamp: selected timestamp from curve
+        """
+        return datetime.utcfromtimestamp(timestamp).strftime(self.date_format)
+
+    def _resetCustomStyle(self):
+        """
+        If Date Inspector created the custom style this method reset the style
+        during the disable process.
+        """
+        for i, curve in enumerate(filter(lambda o: isinstance(o, PlotDataItem),
+                                         self.plot.curves)):
+            if i in self._custom_point_style_init.keys():
+                if self._custom_point_style_init[i]:
+                    curve.setSymbolSize(0)
+                    curve.setSymbol(None)
+
+
+class PointPicker(np.ndarray):
+    """
+    Class use to handle the data with x or y date from curve and checking
+    if the point is near or not (base on the roi) the current mouse point.
+    This class extend the :class:`numpy.ndarray` to handle the numpy array
+    with curve data.
+    """
+    def __new__(cls, *args, **kwargs):
```

Comment:
Just a note on style. Maybe this is a justified use of `__new__`, but IMHO in most cases one can (should) use `__init__` instead for writing higher-level more pythonic code. See https://stackoverflow.com/a/674369